src/modm/platform/comp/stm32/base.hpp.in

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018, Raphael Lehmann
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_STM32_COMP_BASE_HPP
+#define MODM_STM32_COMP_BASE_HPP
+
+/**
+ * @ingroup 	{{partname}}
+ * @defgroup	{{partname}}_comp Comparator
+ */
+namespace modm::platform
+{
+	class CompBase
+	{
+	public:
+		enum class
+		Mode
+		{
+			{% if driver.type in ["stm32-v1.3"] -%}
+			UltraLowPower	= 0b00 << 2,
+			LowPower		= 0b01 << 2,
+			MediumSpeed		= 0b10 << 2,
+			HighSpeed		= 0b11 << 2,
+			{% elif driver.type in ["stm32-tsmc90_cube"] -%}
```

Comment:
These should be renamed in this file: 
https://github.com/modm-io/modm-devices/blob/develop/tools/generator/dfg/stm32/stm_peripherals.py

Would it be a lot of work to check the binary compatiblity of all STM32 Comp devices and add them too?
You'd need to compare the peripheral memory map overview of all devices with this functionality.

Reply:
I have no idea how the different comparator IPs should be grouped, so I don't think a renaming makes sense at the moment:
* stm32-tsmc90_cube
* stm32-tsmc90_h7_cube
* stm32-tsmc90_orca512_cube
* stm32-v1.0
* stm32-v1.2
* stm32-v1.3
* stm32-v1.4
* stm32-v3.4
* stm32-v3.6

I don't think it makes sense to compare the memory maps. I prefer to wait a few weeks and compare the data of the register descriptions from the PDF data sheets.

The memory maps of L43x and F303/F3x8 look very similar, but some bits have other meaning: On L43x the HighSpeed mode is `0b00`, on F303/F3x8 the same bit have to be `0b11` to enable the HighSpeed mode :unamused: 

Data about the configurations of the input- or output- selection (etc.) bits is not present in the memory maps and CMSIS header files.